Src/FluentAssertions/Execution/AssertionScope.cs

```diff
@@ -328,6 +340,15 @@ public void AddReportable(string key, string value)
             contextData.Add(new ContextDataItems.DataItem(key, value, reportable: true, requiresFormatting: false));
         }
 
+        /// <summary>
+        /// Adds some information to the assertion scope that will be included in the message
+        /// that is emitted if an assertion fails. The value is only calculated on failure.
+        /// </summary>
+        public void AddDeferredReportable(string key, Func<string> valueFunc)
```

Comment:
Why not make it an overload of  `AddReportable`?

Reply:
Good point, let me fix that.